cmake.md

```diff
@@ -0,0 +1,24 @@
+# Experimental Build with CMAKE
+
+## Setup
+* Clone the repository. if you have an old version of git, you may need to pass `--recursive` to git.
+* Download Microchip XC16 compiler from http://microchipdeveloper.com/xc16:installation
+* Download the Peripheral Libraries (PLIBS) for "PIC24/dsPIC® DSCs" from http://www.microchip.com/mplab/compilers#pic24
+and install them in the same folder
+* Make sure the directory of the xc16-gcc compiler in in the PATH
+* make sure you have a recent version of CMake - 3.5 or better.
+* You will also have the master branch of aseba cloned
+
+## Build
+* Create a build directory `mkdir build && cd build`
+* invoke cmake : `cmake -D<path/to/aseba>  ..`
+* build: `make`
+
+## Ninja
```

Comment:
I understand that you have a preference, I personally prefer Xcode for example, but I'm not sure that we should inject these preferences here. If people want to change their tooling, that discussion should happen somewhere else. Also, for the user it isn't clear who the authority called "I" here is: the development team, or the (unnamed) person who wrote this commit?

The standard tooling in the Aseba documentation is (as I understand it) basic, very portable cmake + make. This is the same tooling we use in the continuous integration.

If you want to provide instructions for different build systems (make, Ninja, Xcode, MingW32, MSVS), sure, if there are important differences between them.